src/App.tsx

```diff
@@ -70,6 +70,14 @@ class AppMain extends Component {
       flipped: false,
     }
   }
+  componentDidMount() {
+    const parsedQueryString: {
+      amount?: string
+    } = (window.location.search.slice(1) ?? '').split('&').map(o => [o.split('=')[0], o.split('=')[1]]).reduce((acc, [k, v]) => ({ ...acc, [k]: v }), {})
+    if ((parsedQueryString?.amount ?? '').length !== 0 && !Number.isNaN(Number(parsedQueryString?.amount))) {
+      this.setState({ amount: Number(parsedQueryString?.amount) })
+    }
```

Comment:
Modern web comes with `URLSearchParams`:

```suggestion
    const searchParams = new URLSearchParams(location.search)
    const amount = +searchParams.get('amount')
    if (amount) this.setState({ amount })
```